pkg/cmd/dedicated/register/registercluster.go

```diff
@@ -0,0 +1,433 @@
+	machinePoolTaintEffect = "NoExecute"
+	machinePoolTaintValue  = "standard"
+	//machinePoolInstanceType = "m5.2xlarge"
+	machinePoolInstanceType = "r5.xlarge"
+	machinePoolLabelKey     = "bf2.org/kafkaInstanceProfileType"
+	machinePoolLabelValue   = "standard"
+	clusterReadyState       = "ready"
+	fleetshardAddonId       = "kas-fleetshard-operator"
+	strimziAddonId          = "managed-kafka"
+)
+
+func NewRegisterClusterCommand(f *factory.Factory) *cobra.Command {
+	opts := &options{
+		f: f,
+	}
+
+	//TODO add Localizer
+	cmd := &cobra.Command{
+		Use: "register-cluster",
+		// Short: f.Localizer.MustLocalize("registerCluster.cmd.shortDescription"),
+		Short:   "registerCluster.cmd.shortDescription",
+		Long:    "registerCluster.cmd.longDescription",
+		Example: "registerCluster.cmd.example",
+		Args:    cobra.NoArgs,
+		RunE: func(cmd *cobra.Command, _ []string) error {
+			return runRegisterClusterCmd(opts)
+		},
+	}
+
+	// TODO add Localizer and flags
+	// add a flag for clustermgmt url, i.e --cluster-management-api-url, make the flag hidden, default to api.openshift.com
+	// supply customer mgmt access token via a flag, i.e --access-token, make the flag hidden, default to ""
+	flags := kafkaFlagutil.NewFlagSet(cmd, f.Localizer)
+	// this flag will allow the user to pass the cluster id as a flag
+	flags.StringVar(&opts.selectedClusterId, "cluster-id", "", "cluster id")
+	//flags.StringVar(&opts.selectedClusterId, "cluster-id", "", f.Localizer.MustLocalize("registerCluster.flag.clusterId"))
+
+	return cmd
+}
+
+func runRegisterClusterCmd(opts *options) error {
+	getListClusters(opts)
+	if opts.selectedClusterId == "" {
+		runClusterSelectionInteractivePrompt(opts)
+	} else {
+		for _, cluster := range opts.clusterList {
+			if cluster.ID() == opts.selectedClusterId {
+				opts.selectedCluster = cluster
+			}
+		}
+	}
+	getOrCreateMachinePoolList(opts)
+	selectAccessPrivateNetworkInteractivePrompt(opts)
+	registerClusterWithKasFleetManager(opts)
+
+	return nil
+}
+
+func getListClusters(opts *options) error {
+	// ocm client connection
+	conn, err := opts.f.Connection()
+	if err != nil {
+		return err
+	}
+	client, cc, err := conn.API().OCMClustermgmt()
+	if err != nil {
+		return err
+	}
+	defer cc()
+	// TODO deal with pagination, validate clusters -- must be multi AZ and ready.
+	resource := client.Clusters().List()
+	response, err := resource.Send()
+	if err != nil {
+		return err
+	}
+	clusters := response.Items()
+	var cls = []clustersmgmtv1.Cluster{}
+	for _, cluster := range clusters.Slice() {
+		// TODO the cluster must be multiAZ
+		if cluster.State() == clusterReadyState && cluster.MultiAZ() == true {
+			//if cluster.State() == clusterReadyState {
+			cls = append(cls, *cluster)
+		}
+	}
+	opts.clusterList = cls
+	return nil
+}
+
+func runClusterSelectionInteractivePrompt(opts *options) error {
+	// TODO handle in case of empty cluster list, must be cleared up with UX etc.
+	clusterListString := make([]string, 0)
```

Comment:
The name looks bit vague, why we include string here?

Reply:
it is a `clusterList` but in string form. The actual `*option` is:
```
	clusterList                   []clustersmgmtv1.Cluster

```

prompt will only work with strings, 
is there a better way to do this as opposed to how I am handling it currently?